server/src/models/users.js

```diff
@@ -1,6 +1,17 @@
 const { ObjectId } = require('mongodb');
 const mongoose = require('mongoose');
 
+if (process.env.NODE_ENV !== "production") {
+    const dotenv = require("dotenv")
+    dotenv.config()
+}
+
+const auth0ApiUrl = `https://${process.env.AUTH0_DOMAIN}/api/v2`
```

Comment:
These DB models are typically left to only to hold DB schema - so adding in Auth0 to this is slightly off from normal - was this recommended somewhere @benhsm ?

Reply:
No, this I didn't see this recommended anywhere. For our context and situation, I view this as necessary because of the requirement that Auth0 user data and the application's user data have to be kept in sync. The code for handling calls to the Auth0 /user endpoints has be either in the route handlers or here where the schema is defined. I opted for the latter in part because it seems to result in less duplication of logic, but also because this is a way to represent a real data-integrity constraint. From what I can see, there should be no circumstances in which an Auth0 user entity exists without corresponding application user, or vice versa, and this hook is an attempt to enforce that at the data layer.